wherescape/connectors/gsheet/gsheets_wrapper.py

```diff
@@ -0,0 +1,272 @@
+        if self.spreadsheet is not None:
+            return self.spreadsheet
+        else:
+            self.set_spreadsheet(url, name)
+            return self.spreadsheet
+
+    def set_worksheet(self, title: str = None):
+        """
+        Sets the worksheet based on the given title.
+        A spreadsheet needs to have already been set.
+
+        Args:
+        - title (str) : name of spreadsheet default value is standard among most
+            spreadsheet applications
+        """
+        try:
+            if self.spreadsheet is None:
+                logging.info("No name was provided. Using the first sheet")
+                self.spreadsheet = self.worksheet.get_worksheet(0)
+            else:
+                self.worksheet = self.spreadsheet.worksheet(title)
+        except WorksheetNotFound as notFound:
+            logging.warning("No worksheet was found")
+            raise notFound
+
+        logging.info(f"worksheet found")
+
+    def get_worksheet(self, title: str = "") -> Worksheet:
+        """
+        Getter for worksheet. Calls setter if not already present.
+
+        Params:
+        - title (str) : title of the worksheet
+
+        Returns:
+        - Worksheet
+        """
+        if self.worksheet:
+            pass
+        elif title:
+            self.set_worksheet(title)
+        else:
+            self.set_worksheet()
+        return self.worksheet
+
+    def set_content(self, range: str = None):
+        """
+        Retreives content from gsheet. removes empty rows.
+        """
+        if range:
+            try:
+                content = self.worksheet.get(range, )
+            except APIError as e:
+                logging.error(f"Invalid range: {range}")
+                raise e
+        else:
+            content = self.worksheet.get_all_values()
+        self.content = remove_empty_rows_and_columns(content)
+
+    def get_content(self, range: str = None) -> list:
+        """
+        Getter for content. Calls setter if not already present.
+
+        Returns:
+        - List of lists containing the content of the sheet.
+        """
+        if self.content is None:
+            self.set_content(range)
+        return self.content
+
+    def set_header(self, no_header: str = None, header_range: str = None):
+        """
+        Creates the header for the content. Takes it from content if no_header is False.
+
+        Params:
+        - no_header (str): (optional) input of args.no_header
+        - header_Range (str): (optional) range of the header example: "A1:B5"
+        """
+        # Set content if none is set yet.
+        if self.content is None:
+            self.set_content()
+        
+        if not no_header:
+            if header_range:
+                try:
+                    row = self.worksheet.get(header_range)[0]
+                    self.header = ["column_" + str(i + 1) if value == "" else value for i, value in enumerate(row)]
+                    self.content.pop(0)
```

Comment:
it is because the header is still considered included in content. There is the issue that it assumes the first row of the content is the header, which could probably be improved upon, but I don't have an immediate idea of a solution. Might be a decent small ticket for improvement for later